doc/develop/test/twister.rst

```diff
@@ -1315,6 +1315,38 @@ Using Single Board For Multiple Variants
       runner: nrfjprog
       serial: /dev/ttyACM1
 
+Scripts Usage for platform
+++++++++++++++++++++++++++
+
+Below scripts are supported
+
+* pre_script: <path to pre script>
+
+* post_script: <path to pre script>
+
+* post_flash_script: <path to pre script>
+
+each of above scripts can have timeout defined as below in hardwar map file by script_param
+
+.. code-block:: yaml
+
+    - connected: false
+      id: 0229000005d9ebc600000000000000000000000097969905
+      platform: mimxrt1060_evk
+      probe_id: 000609301751
+      product: DAPLink CMSIS-DAP
+      runner: jlink
+      serial: null
+      pre_script: pre_scripts.sh
+      post_script: post_scripts.sh
+      post_flash_script: post_flash_scripts.sh
+      script_param:
+        "pre_script_timeout":  100
+        "post_script_timeout": 100
+        "post_flash_timeout":  100
+
+also a default patameter <testinstance.name> will be pass as fist parameter for scripts
```

Comment:
@hakehuang, to be on the same page: why you need the instance name to pass to scripts ?

btw. its `platform` prefix is not 'normalized' (it is in HWMv2 format), whereas the `path` component might be excluded with `--no-detailed-test-id` option.

Reply:
we have a platform, for security reasons, it need pre-load different security blobs to board first, before run test. @golowanow . 


> btw. it is `platform/suite_path/suite_name` where `platform` prefix is not 'normalized' (it is in HWMv2 format), whereas `suite_path` might be excluded with `--no-detailed-test-id` option.

it is not an issue at my case, as per_script is always defined in platform level, so we know which platform it is running already. and suite_path is not need usually, as pre_script does not always go with the build but run

Reply:
> it is not an issue at my case, as per_script is always defined in platform level, so we know which platform it is running already. and suite_path is not need usually, as pre_script does not always go with the build but run

so in your use case the parameter it is just for `suite_name` actually ?

well, I think it is a quite frequent situation when the same flash- script is re-used with different platforms/variants (currently that can be solved creating links to the same script file with different names), and the better and general solution would be to call these scripts with a basic set of expected parameters (like what Twister has `--platform`, `--scenario`, etc.) in addition to the command line from the map file the script is called with.